### PR TITLE
Creation of document-generator repo

### PR DESCRIPTION
**Pull request by Mathsh**
Added basic .gitignore, start.sh, pom.xml, MakeFile and ReadMe for new document-generator Repo

Resolves SFA-425

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+document-generator
```
Comment (BSaunders95) [inline README.md]: This should be prefixed with a '#' to format it as a title

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                        </configuration>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-jar-plugin</artifactId>
+                        <version>${maven-jar-plugin.version}</version>
+                        <configuration>
+                            <archive>
+                                <manifest>
+                                    <mainClass>uk.gov.companieshouse.session.Harness</mainClass>
```
Comment (BSaunders95) [inline pom.xml]: This class does not exist in the project

File: README.md (hunk 1 above)
Comment (Mathsh) [reply README.md]: Have added

File: start.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    source ~/.chs_env/global_env
+
+    PORT="${DOCUMENT_GENERATION_SERVICE_PORT}"
+else
+    PORT="$1"
+    CONFIG_URL="$2"
+    ENVIRONMENT="$3"
+    APP_NAME="$4"
+
+    source /etc/profile
```
Comment (BSaunders95) [inline start.sh]: Does this file need to be sourced?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+package:
+    mvn versions:set -DnewVersion=$(version) -DgenerateBackupPoms=false
+    mvn package -DskipTests=true
+
+.PHONY: dist
+dist: clean package
+
+.PHONY: publish
+publish:
+    mvn jar:jar deploy:deploy -DpublishRepo=$(artifactory_publish)
```
Comment (jsimons1989) [inline Makefile]: Will this work with the submodules?

Comment (Mathsh) [reply Makefile]: Not 100% sure yet. I'll remove for now till we can test

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <build>
+                <finalName>${project.artifactId}-${project.version}</finalName>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <version>${maven-compiler-plugin.version}</version>
```
Comment (jsimons1989) [inline pom.xml]: maven-compiler-plugin.version isn't specified

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                            <fork>true</fork>
+                            <meminitial>128m</meminitial>
+                            <encoding>${project.build.sourceEncoding}</encoding>
+                            <maxmem>512m</maxmem>
+                        </configuration>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-jar-plugin</artifactId>
+                        <version>${maven-jar-plugin.version}</version>
```
Comment (jsimons1989) [inline pom.xml]: maven-jar-plugin.version isn't specified

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>${maven-surefire-plugin.version}</version>
```
Comment (jsimons1989) [inline pom.xml]: maven-surefire-plugin.version isn't specified

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+                        <dependencies>
+                            <dependency>
+                                <groupId>org.junit.platform</groupId>
+                                <artifactId>junit-platform-surefire-provider</artifactId>
+                                <version>${junit-platform-surefire-provider.version}</version>
```
Comment (jsimons1989) [inline pom.xml]: junit-platform-surefire-provider.version isn't specified

File: pom.xml (hunk 5 above)
Comment (Mathsh) [reply pom.xml]: Has been added

File: pom.xml (hunk 6 above)
Comment (Mathsh) [reply pom.xml]: Has been added

File: pom.xml (hunk 7 above)
Comment (Mathsh) [reply pom.xml]: Has been added

File: pom.xml (hunk 8 above)
Comment (Mathsh) [reply pom.xml]: Has been added

**Pull request merged**